src/na/device/App.cpp

```diff
@@ -0,0 +1,385 @@
+/*
+ * Copyright (c) 2023 - 2025 Chair for Design Automation, TUM
+ * Copyright (c) 2025 Munich Quantum Software Company GmbH
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier: MIT
+ *
+ * Licensed under the MIT License
+ */
+
+#include "na/device/Generator.hpp"
+#include "na/device/device.pb.h"
+#include "spdlog/spdlog.h"
+
+#include <cstddef>
+#include <cstdint>
+#include <exception>
+#include <iostream>
+#include <optional>
+#include <stdexcept>
+#include <string>
+#include <utility>
+#include <vector>
+
+namespace {
+/**
+ * Prints the usage information for the command line tool.
+ * @param programName is the name of the program executable.
+ */
+auto printUsage(const std::string& programName) -> void {
+  std::cout
+      << "Generator of a header file from a JSON file for neutral atom QDMI\n"
+         "implementation.\n"
```

Comment:
```suggestion
      << "Generator for turning neutral atom computer JSON specifications into header files to be used as part of a neutral atom QDMI device implementation.\n"
```
a bit verbose, but maybe slightly better.